### DISCUSSION TITLE
Create config failing with new error

### DISCUSSION THREAD

**Discussion by alexwillingham98** [thoughtspot/cs_tools#188] [Q&A] [answered]
┌───────────────────── Traceback (most recent call last) ─────────────────────┐
│ C:\Users\VssAdministrator\AppData\Roaming\cs_tools\.cs_tools\Lib\site-packa │
│ ges\cs_tools\cli\commands\config.py:98 in create                            │
│                                                                             │
│    95 │                                                                     │
│    96 │   try:                                                              │
│    97 │   │   log.info("Checking supplied configuration..")                 │
│ >  98 │   │   ts.login()                                                    │
│    99 │                                                                     │
│   100 │   except errors.AuthenticationError as e:                           │
│   101 │   │   log.debug(e, exc_info=True)                                   │
│                                                                             │
│ C:\Users\VssAdministrator\AppData\Roaming\cs_tools\.cs_tools\Lib\site-packa │
│ ges\cs_tools\thoughtspot.py:222 in login                                    │
│                                                                             │
│   219 │   │   │   │   login_info.pop("password")                            │
│   220 │   │                                                                 │
│   221 │   │   try:                                                          │
│ > 222 │   │   │   log.debug(f"SESSION CONTEXT\n{json.dumps(self.session_con │
│   223 │   │   except errors.NoSessionEstablished:                           │
│   224 │   │   │   for method, r in attempted_auth_method.items():           │
│   225 │   │   │   │   log.info(f"Attempted  {method.title().replace('_', '  │
│                                                                             │
│ C:\Users\VssAdministrator\AppData\Roaming\cs_tools\.cs_tools\Lib\site-packa │
│ ges\pydantic\main.py:421 in model_dump                                      │
│                                                                             │
│    418 │   │   Returns:                                                     │
│    419 │   │   │   A dictionary representation of the model.                │
│    420 │   │   """                                                          │
│ >  421 │   │   return self.__pydantic_serializer__.to_python(               │
│    422 │   │   │   self,                                                    │
│    423 │   │   │   mode=mode,                                               │
│    424 │   │   │   by_alias=by_alias,                                       │
└─────────────────────────────────────────────────────────────────────────────┘
PydanticSerializationError: Unable to serialize unknown type: <class           
'pydantic.networks.AnyUrl'>

This is the command I run - .\cs_tools.exe config create --config main --username '$(CS_TOOLS_USERNAME)' --password '$(CS_TOOLS_PASSWORD)'  --url '$(TS_URL)' --disable_sso --verbose --default

**Comment by boonhapus** [comment]
😄 you're so fast @alexwillingham98 ! **[Our own CI failed as well](https://github.com/thoughtspot/cs_tools/actions/runs/11947085459/job/33302496297)** due to Pydantic's update overnight to v2.10 Issues.

**Reply by boonhapus** [reply]
I'll fix the Pydantic version to 2.9 and push and new CS Tools version later today.

**Reply by alexwillingham98** [reply]
I have a daily job setup in devops so just happened to notice it failed this morning 😉

**Reply by boonhapus** [reply, marked as answer]
@alexwillingham98 Just test -- this should be good to go again for the next CI run.

https://github.com/thoughtspot/cs_tools/actions/runs/11964943017

**Reply by alexwillingham98** [reply]
The job is passing now, thank you!
